src/ripple/app/paths/PathRequest.cpp

```diff
@@ -175,70 +166,69 @@ void PathRequest::updateComplete ()
 bool PathRequest::isValid (RippleLineCache::ref crCache)
 {
     ScopedLockType sl (mLock);
-    bValid = raSrcAccount && raDstAccount &&
-            saDstAmount > zero;
-    auto const& lrLedger = crCache->getLedger ();
+    if (! raSrcAccount || ! raDstAccount)
+        return false;
 
-    if (bValid)
+    if (! convert_all_ && (saSendMax || saDstAmount <= zero))
     {
-        if (! crCache->getLedger()->exists(
-                keylet::account(*raSrcAccount)))
-        {
-            // no source account
-            bValid = false;
-            jvStatus = rpcError (rpcSRC_ACT_NOT_FOUND);
-        }
+        // if send max specified, dst amt must be -1
+        jvStatus = rpcError(rpcDST_AMT_MALFORMED);
+        return false;
     }
 
-    if (bValid)
+    if (! crCache->getLedger()->exists(
+        keylet::account(*raSrcAccount)))
     {
-        auto const sleDest = crCache->getLedger()->read(
-            keylet::account(*raDstAccount));
+        // source account does not exist
```

Comment:
Shouldn't you check somewhere to see if `usDestCurrID` has more than zero elements in it?